crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1661,4 +1721,120 @@ where
         )
         .await
     }
+
+    /// Given a certificate, execute the cert on remote validators (and preferentially on the
+    /// signers of the cert who are guaranteed to be able to process it immediately) until we
+    /// receive f+1 identical SignedTransactionEffects - at this point we know we have the
+    /// true effects for the cert, because of f+1 validators, at least 1 must be honest.
+    ///
+    /// It is assumed that this method will not be called by any of the signers of the cert, since
+    /// they can simply execute the cert locally and compute their own effects.
```

Comment:
I think here we somehow confuse how full node get to learn effects vs. how validators do. 

Remember: full nodes do not participate in consensus. So they learn of the input objects for a shared object transaction only through certified effects. However, a validator learns of the shared object version numbers by participating in consensus, sending a cert not seen sequenced through consensus, and waiting for its sequencing to assign object versions.

As a result even the validators that have not signed the cert should be able to process the cert without waiting for an effects cert / quorum.

Reply:
No confusion - the purpose of this PR is to deal with missing parent certificates. The only validators that are guaranteed not to have missing parent certificates are the signers of the cert. That's all I mean here - please let me know if you have suggestions on how to make the comment clearer.

Reply:
(The need for the effects is not to sequence shared objects, but to find out what the missing certs are so that they can be fetched)

Reply:
Ok, got it!